e2e/_suites/fleet/features/fleet_server.feature

```diff
@@ -0,0 +1,19 @@
+@fleet_server
+Feature: Fleet Server
+  Scenarios for Fleet Server, where an Elasticseach and a Kibana instances are already provisioned,
+  so that the Agent is able to communicate with them
+
+@start-fleet-server
+Scenario Outline: Deploying the <os> fleet-server agent enables Fleet Server
```

Comment:
This is surely because of my lack of understanding of how fleet-server works, but I don't find this phrasing very clear: there is no such a thing as "fleet-server agent", right? There is an Elastic Agent that can start a Fleet Server process: shouldn't this be in the scenario definition?

Since "Fleet" is such a ubiquitous term, it might be worth to be more explicit every time is it used: Is it Fleet API, Fleet UI, etc.
Related: since this spec is meant to be semi-formal, and we put a lot of effort in standarisation elsewhere (linting, etc), It would be good to define components canonically. Eg., always "Elastic Agent", and not a combination of "Elastic Agent", "Agent", "the agent", etc.

Reply:
**I strongly agree with this. 💯**

What about sending the refactors in a follow-up? Let's create an issue to standardise the names across Fleet test suite. Maybe @EricDavisX can help with the wording.

Reply:
Related to the name, yes. It could be rephrased. Let me add a suggestion.

Reply:
Another note about this pr vs a separate one...  we have a new feature file which is helpful, but to me, it is mostly helpful in confirming we have the code right here, with a straightforward usage.  We'll need to adapt (now or in a separate PR) all of the:
  Given a "<os>" agent is deployed to Fleet with "tar" installer
to
  When a "<os>" agent is deployed to Fleet with "tar" installer in fleet-server mode
...because any non-Fleet server usage (in Fleet) will not be supported.  The only non-Fleet-Server usage will be as a stand-alone Agent mode.

So, all the following feature files will need review and update, at least in some way (listing them out explicitly):
e2e/_suites/fleet/features/agent_endpoint_integration.feature
e2e/_suites/fleet/features/backend_processes.feature
e2e/_suites/fleet/features/linux_integration.feature
e2e/_suites/fleet/features/fleet_mode_agent.feature

The stand_alone_agent file is the exception:
e2e/_suites/fleet/features/stand_alone_agent.feature
 - we have opportunity there to improve the stand-alone test to include Docker usage with Fleet Server.  And it may be a great and easy way to spin up a 2nd Agent to connect to the first (the first Agent also running the Fleet Server, and the 2nd being a 'normal' Agent)

So, knowing this now, I'm not sure if we want to keep this first PR small and merge it so we have some passing scenario knowing the others will fail, or if we'll want to push forward with slightly larger impact across more files after we see it working.